test/rules/typeof-compare/test.js.lint

```diff
@@ -0,0 +1,38 @@
+var testVariable = 123;
+
+function testFunction() {
+    if (typeof x === 'null') {
+        ~~~~~~~~~~~~~~~~~~~      [typeof must be compared to correct value]
+        console.log("called");
+    }
+    const x = typeof foo === 'undefied'
+              ~~~~~~~~~~~~~~~~~~~~~~~~~      [typeof must be compared to correct value]
+}
+
+console.log(typeof window == 'object ')
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~      [typeof must be compared to correct value]
+
+const string = "string";
+typeof bar === string
```

Comment:
needs more positive tests. Currently only tests against a string variable. Needs to test against directly against a string, against a template string, and a template string with a replacement variable. Also exercise different types of valid types